addons/binding/org.openhab.binding.exec/README.md

```diff
@@ -62,3 +62,86 @@ Switch APCRunning { channel="exec:command:apc:run"}
+The folder structure is for an installation with apt-get, for a different installation setup look [here](http://docs.openhab.org/installation/linux.html#file-locations) or search for them in your drive.
+
+First we need to check if the user openhab is able to execute the command we want to execute. The first command is executet as the user we are logged in the second as user openhab. If you get an error or an massages which indicates the command does not execute properly as user openhab then try to google the needed permissions and set them for the user openhab. More details [here](https://community.openhab.org/t/2-openhab2-rpi-system-temperature-chart-with-persistence/35182) and [here](https://community.openhab.org/t/openhab-sudo-exec-binding/34988).
+
+```
+cat /sys/class/thermal/thermal_zone0/temp
+48312
+sudo -u openhab cat /sys/class/thermal/thermal_zone0/temp
+47774
+```
+
+As the command got executed as user openhab we can proceed to set up our OpenHAB.
+
+First we need a **Thing** file which configures the command line call to execute.  
+```
+sudo nano /etc/openhab2/things/exec.things
+```
+```
+Thing exec:command:cpuTemp [
+        command="cat /sys/class/thermal/thermal_zone0/temp",
+        interval=10,
+        autorun=false]
+```
+
+Then we need an **Items** file to store hte string we get back from the execution and to stored the transformed and divided temperature as number.
+
+```
+sudo nano /etc/openhab2/items/SysTemp.items
+```
+```
+Number System_Temperature_CPU "Temperature CPU [%.1f °C]"
+
+// Output of command line execution 
+String cpuTemp_out { channel="exec:command:cpuTemp:output" }
+```
+
+Then we need a **Sitemap** file to configure the site which will be displayed. **It has to have the same name as the sitemap inside the file.**
+```
+sudo nano /etc/openhab2/sitemaps/SysTemp.sitemap
+```
+```
+// Name of file and name of sitemap has to be the same
+sitemap SysTemp label="System Temperature RPI"
+{
+        Frame {
+            Text item=System_Temperature_CPU
+        }
+}
+```
+
+Now we need a **Rule** file which is triggered when the returned string of our execution changes and then transforms string to a number, divide it and also log the output of the execution.
+```
+sudo nano /etc/openhab2/rules/SysTemp.rules
+```
+```
+rule "System CPU Temperature"
+  when
+     Item cpuTemp_out received update
+  then
+  
+      System_Temperature_CPU.postUpdate( 
+            (Integer::parseInt(cpuTemp_out.state.toString) as Number ) 
+            /1000 
+      )
+      
+      logInfo("CPU Temp", cpuTemp_out.state.toString.trim )
+end
+
+```
+
+The logging massages can be viewed in the Karaf console have a closer look [here](http://docs.openhab.org/administration/console.html) for more information.
+
+If you like to chart the values follow the [Tutorial](https://community.openhab.org/t/2-openhab2-rpi-system-temperature-chart-with-persistence/35182)
```

Comment:
I doubt whether this advertising and crediting is needed and appropriate for the readme I think it's safe to leave it out.

Reply:
Hello Martin, 

first of all, thank you for your review.

I think as this full example addresses people which just start to get in the topic, it is a good idea to point out where they can find apropriate information on how to set up thinks.
As i  am also just started to work into this and find it quite bothersome to search the bits and peaces, i thought i would contribute my gatherings.

I changed the sentence and the link to point out to the forum. hopefully this is ok.

I would like to have a full example to every binding which not expect that i am allready a pro user and directly grasp everything which is stated. 

And as the manual for the bindings seems to be de facto the readme i think it is the place to point out how thinks work and where you can continue.

best regards
Josua

Reply:
I agree to remove links